src/vs/platform/windows/electron-main/windowImpl.ts

```diff
@@ -287,11 +287,26 @@ export abstract class BaseWindow extends Disposable implements IBaseWindow {
 		return !!this.documentEdited;
 	}
 
-	focus(options?: { force: boolean }): void {
-		if (isMacintosh && options?.force) {
+	focus(options?: { force: boolean; notify: boolean }): void {
```

Comment:
From the signature it seems I can both forcefully focus and notify, but from the implementation that is not the case? Should we either allow that or make the signature be either or?

Reply:
I think the nice way to do that would by changing this arg to `inBackground: 'bringToFront' | 'notify' | 'none'` or an equivalent enum. In #247609 I thought that's not what you wanted but I'm happy to make the change if you think it's a good one :)

Reply:
Yes I like that!

But how do we really distinguish `force: true` from `force: false`. I mean in the end even `force: false` can still focus, so is it rather:

```ts
focus(mode?: 'force' | 'notify')
```

?

Reply:
I like enums for this case because they're easier to document, especially since this flag gets passed around through several interfaces, so maybe something like

```ts
const enum FocusMode {
	/** (Default) Transfer focus to the target window when if the editor is focused */
	Transfer,
	/** Transfer focus to the target window when if the editor is focused, otherwise notify the user that the app has activity (macOS/Windows only) */
	Notify,
	/** Force the window to be focused, even if the window is not currently focused */
	Force,
}

focus(mode?: FocusMode) 
```

Reply:
Yeah lets try that, looks ok to me!